pkg/gcptarget/gcptarget.go

```diff
@@ -0,0 +1,117 @@
+)
+
+// Name is the target name for GCPTarget
+const Name = "validation.gcp.forsetisecurity.org"
+
+// GCPTarget is the constraint framework target for FCV
+type GCPTarget struct {
+}
+
+var _ client.TargetHandler = &GCPTarget{}
+
+// New returns a new GCPTarget
+func New() *GCPTarget {
+	return &GCPTarget{}
+}
+
+// MatchSchema implements client.MatchSchemaProvider
+func (g *GCPTarget) MatchSchema() apiextensions.JSONSchemaProps {
+	return apiextensions.JSONSchemaProps{
+		Properties: map[string]apiextensions.JSONSchemaProps{
+			"target": {
+				Type: "array",
+				Items: &apiextensions.JSONSchemaPropsOrArray{
+					Schema: &apiextensions.JSONSchemaProps{
+						Type: "string",
+					},
+				},
+			},
+			"exclude": {
+				Type: "array",
+				Items: &apiextensions.JSONSchemaPropsOrArray{
+					Schema: &apiextensions.JSONSchemaProps{
+						Type: "string",
+					},
+				},
+			},
+		},
+	}
+}
+
+// GetName implements client.TargetHandler
+func (g *GCPTarget) GetName() string {
+	// Done
+	return Name
+}
+
+// Library implements client.TargetHandler
+func (g *GCPTarget) Library() *template.Template {
+	// TODO - lib impl
+	return libraryTemplate
+}
+
+// ProcessData implements client.TargetHandler
+func (g *GCPTarget) ProcessData(obj interface{}) (bool, string, interface{}, error) {
+	return false, "", nil, errors.Errorf("Storing data for referential constraint eval is not supported at this time.")
+}
+
+// HandleReview implements client.TargetHandler
+func (g *GCPTarget) HandleReview(obj interface{}) (bool, interface{}, error) {
+	switch asset := obj.(type) {
+	case *validator.Asset:
+		return g.handleAsset(asset)
+	case map[string]interface{}:
+		return true, asset, nil
+	}
+
+	return false, nil, nil
+}
+
+// handleAsset handles input from FCV assets as received via the gRPC interface.
+func (g *GCPTarget) handleAsset(asset *validator.Asset) (bool, interface{}, error) {
+	if asset.Resource == nil {
+		return false, nil, errors.Errorf("forseti asset's resource field is nil %s", asset)
+	}
+	asset2.CleanStructValue(asset.Resource.Data)
+	m := &jsonpb.Marshaler{
+		OrigName: true,
+	}
+	var buf bytes.Buffer
+	if err := m.Marshal(&buf, asset); err != nil {
+		return false, nil, errors.Wrapf(err, "marshalling to json with asset %s: %v", asset.Name, asset)
+	}
+	var f interface{}
+	err := json.Unmarshal(buf.Bytes(), &f)
+	if err != nil {
+		return false, nil, errors.Wrapf(err, "marshalling from json with asset %s: %v", asset.Name, asset)
+	}
+	return true, f, nil
+}
+
+// HandleViolation implements client.TargetHandler
+func (g *GCPTarget) HandleViolation(result *types.Result) error {
+	// Done? - this mutates the result after it's provided, we may have to add metadata
+	// to the result
+	return nil
+}
+
+// ValidateConstraint implements client.TargetHandler
+func (g *GCPTarget) ValidateConstraint(constraint *unstructured.Unstructured) error {
+	// TODO: figure out what semantic validation we need here, maybe validate the match criteria
```

Comment:
The only thing I can think of here is to make sure the strings are well formed (e.g., non-`*` characters followed by a `*`)

Reply:
added regex compilation to this